src/compiler/utilities.ts

```diff
@@ -2599,14 +2599,20 @@ namespace ts {
     }
 
     export function getEffectiveJSDocHost(node: Node): Node | undefined {
-        const host = getJSDocHost(node);
-        const decl = getSourceOfDefaultedAssignment(host) ||
-            getSourceOfAssignment(host) ||
-            getSingleInitializerOfVariableStatementOrPropertyDeclaration(host) ||
-            getSingleVariableOfVariableStatement(host) ||
-            getNestedModuleDeclaration(host) ||
-            host;
-        return decl;
+        const jsDoc = findAncestor(node.parent, isJSDoc);
+        if (jsDoc === undefined) {
+            return undefined;
+        }
+
+        const host = jsDoc.parent;
+        if (host.jsDoc && jsDoc === lastOrUndefined(host.jsDoc)) {
+            return getSourceOfDefaultedAssignment(host)
+                || getSourceOfAssignment(host)
+                || getSingleInitializerOfVariableStatementOrPropertyDeclaration(host)
+                || getSingleVariableOfVariableStatement(host)
+                || getNestedModuleDeclaration(host)
+                || host;
+        }
     }
 
     /** Use getEffectiveJSDocHost if you additionally need to look for jsdoc on parent nodes, like assignments.  */

```

Comment:
can you also look through uses of `getJSDocHost` and see what happens if we get rid of `checkDefined` and instead return `undefined` sometimes? Clearly there are unhosted jsdocs.

(Unfortunately this will be a change to the public API too.)

Reply:
I was able to make the changes myself, so I think getJSDocHost should instead have the check that you added in this PR and should add undefined to its return type.

Reply:
@sandersn Thanks for the review 👍🏻. I wanted to add these changes to `getJSDocHost`, however, I had one _failed_ test, so I decided to extend `getEffectiveJSDocHost` which is used in `checkJSDocAugmentsTag`. 

The following test fails

https://github.com/microsoft/TypeScript/blob/0d8a868e91ce5ae2fe0c26de7dbf7d6f7120d17f/tests/cases/conformance/jsdoc/typedefTagTypeResolution.ts#L17-L18

```shell
error TS2322: Type 'number' is not assignable to type 'U'.
  'U' could be instantiated with an arbitrary type which could be unrelated to 'number'.
```
Not sure if these changes are acceptable for all cases where `getJSDocHost` is used.

Reply:
You worked around this in your latest commit, right?

Reply:
@sandersn Yes.